creme/creme_core/gui/menu.py

```diff
@@ -239,6 +251,143 @@ def url(self):
+            )
+
+        return get_template(template_name)
+
+    def get_context(self, request):
+        label = self.render_label(request)
+        user = request['user']
+
+        try:
+            self.check_permissions(user)
+            permission_error = ''
+        except PermissionDenied as e:
+            permission_error = str(e)
+
+        return {
+            'label': label,
+            'permission_error': permission_error,
+            'template_name': self.template_name
+        }
+
+    def render(self, request):
+        context = self.get_context(request)
+        return self.get_template(context).render({
+            'user': request['user'],
+            'request': request,
+            'entry': context,
+        })
+
+
+class ActionEntry(TemplateEntry):
+    template_name = 'creme_core/menu/action.html'
+
+    form_class = menu_forms.ActionEntryForm
+
+    action: str = 'redirect'
+    url_name: str = ''
+
+    icon_name: str = None
+    icon_title: str = None
+    classes: Sequence[str] = ()
+    description: str = ''
+
+    @property
+    def url(self) -> str:
+        url_name = self.url_name
+        return reverse(url_name) if url_name else ''
+
+    def _get_prop(self, name, context):
+        prop = getattr(self, f'get_{name}', None)
+        return prop(context) if callable(prop) else getattr(self, name, None)
+
+    def _get_icon(self, name, title, user) -> BaseIcon:
+        if not self.icon_name:
+            return
+
+        theme = user.theme_info[0]
+        return get_icon_by_name(
+            name=name, label=title, theme=theme,
+            size_px=get_icon_size_px(theme=theme, size='header-menu'),
+        )
+
+    def get_url(self, context):
+        return self.url
+
+    def get_context(self, request):
+        user = request['user']
+
+        entry = super().get_context(request)
+        entry['action'] = self.get_action_context(entry, user)
+
+        return entry
+
+    def get_action_context(self, context, user):
+        prop = partial(self._get_prop, context=context)
+        description = prop('description') or prop('label')
+        permission_error = context.get('permission_error')
+        icon_title = permission_error or prop('icon_title') or description
+        icon_name = prop('icon_name')
+
+        return {
+            'id': self.action,
+            'url': prop('url'),
+            'icon_name': icon_name,
+            'icon_title': icon_title,
+            'icon': self._get_icon(icon_name, icon_title, user=user),
+            'classes': prop('classes'),
+            'description': description,
+            'props': self.get_action_props(context),
+        }
+
+    def get_action_props(self, context):
+        return {
+            "data": self.get_action_data(context),
+            "options": self.get_action_options(context),
+        }
+
+    def get_action_data(self, context):
+        return {}
+
+    def get_action_options(self, context):
+        return {}
```

Comment:
Documentation plz ; what's the difference between data, options & props? No idea, I must read the template..